exercises/difference-of-squares/example.sml

```diff
@@ -1,11 +1,8 @@
 local
-  val identity = fn x => x
-  fun range n = List.tabulate (n, identity)
   fun square n = n * n
-  val sum = foldl (op +) 0
-  fun inc n = n + 1
+  fun sum n = n * (n + 1) div 2
 in
-  val squareOfSum = square o sum o range o inc
-  fun sumOfSquares n = sum (map square ((range o inc) n))
+  val squareOfSum = square o sum
+  fun sumOfSquares n = (sum n) * (2 * n + 1) div 3
```

Comment:
I see that you split the closed form across two functions: `sum` and `sumOfSquares`, could they just be combined into one? Took me a minute to recognize the closed form since it was split.